src/HotChocolate/Data/src/Data/Projections/Expressions/Optimizers/QueryableFilterProjectionOptimizer.cs

```diff
@@ -32,17 +32,18 @@ public class QueryableFilterProjectionOptimizer : IProjectionOptimizer
         resolverPipeline = WrappedPipeline(resolverPipeline);
 
         var compiledSelection = new Selection(
-            context.GetNextId(),
+            selection.Id,
             context.Type,
             selection.Field,
             selection.Field.Type,
             selection.SyntaxNode,
             selection.ResponseName,
-            resolverPipeline,
-            arguments: selection.Arguments,
-            isInternal: false);
+            SelectionExecutionStrategy.Default,
+            selection.Arguments,
+            // TODO I think i need to have access to the include conditions here
```

Comment:
@michaelstaib  i think to clone the selection i need to have access to the include conditions. otherwise this will not work.